### PR TITLE
fix: multiple xblock allowed

### PR DESCRIPTION
**Pull request by johnvente**
## Description

We couldn't add more than one XBlock because only the first one worked. This fix allows us to add as many XBlocks as we want

**STUDIO**

BEFORE         | NOW
:-------------------------:|:-------------------------:
  |  

**LMS**

lms-image

### HOW TO TEST IT
**[tutor]**
clone this branch in your  requirements

**1.** 
```
 cd /env/build/openedx/requirements/xblock-imagesgallery && git clone -b jv/fix-multiple-xblock git@github.com:eduNEXT/xblock-imagesgallery.git
 ```

- -----  lms -------
1. tutor dev exec lms bash
2. pip install -e /openedx/requirements/xblock-imagesgallery
3. exit 

- -----  cms -------
1. tutor dev exec cms bash
2. pip install -e /openedx/requirements/xblock-imagesgallery
3. exit 

**2.** tutor config save

**3.** tutor dev restart

configure the XBlock as[ here](https://github.com/eduNEXT/xblock-imagesgallery/pull/7)

**4.**  Try to add some XBlocks type **imagesgallery**

**Merge checklist:**
Check off if complete *or* not applicable:
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Fixup commits are squashed away
- [ ] Unit tests added/updated
- [ ] Manual testing instructions provided
- [ ] Noted any: Concerns, dependencies, migration issues, deadlines, tickets

### CODE REVIEW

File: imagesgallery/imagesgallery.py [hunk 1]
```diff
@@ -65,6 +65,21 @@ def resource_string(self, path):
...
+        file_content = ''
+        BASE_DIR = os.path.abspath(os.path.dirname(__file__))
+        file_path = os.path.join(BASE_DIR, path)
+
+        try:
+            with open(file_path, 'r') as file_data:
+                file_content = file_data.read()
+
+        except FileNotFoundError as e:
+            print(f"FileNotFoundError: {e}")
```
Comment (Ian2012) [inline imagesgallery/imagesgallery.py]: I guess this error will not be retrieved to users, so you should update this to use the logger

**Review by Ian2012 — APPROVED** [review]
Just one nit

File: imagesgallery/imagesgallery.py [hunk 2]
```diff
@@ -74,22 +89,57 @@ def student_view(self, context=None):
         if context:
             pass  # TO-DO: do something based on the context.
 
-        # Remove the bundle default generated by webpack
-        script_to_remove = r'<script defer="defer" src="bundle.js"></script>'
+        xblock_id = self.block_id
+        # Split the string by the @ symbol and get the last part
+        # XBlock IDs typically follow this format: block-v1:{course_name}+type@{xblockname}+block@{id_xblock}
+        # This code extracts the last part of the string, which is {id_xblock}
+        xblock_id_splitted = xblock_id.split('@')[-1]
```
Comment (BryanttV) [inline imagesgallery/imagesgallery.py]: You can obtain the `block_id` by accessing `self.scope_ids.usage_id.block_id`, and then we can remove the `split`

**Review by nandodev-net — APPROVED** [review]
multiple instances of Images Gallery work seamlessly within the same unit.

File: imagesgallery/imagesgallery.py (hunk 2 above)
Comment (johnvente) [reply imagesgallery/imagesgallery.py]: Thanks! this is better

File: imagesgallery/imagesgallery.py (hunk 1 above)
Comment (johnvente) [reply imagesgallery/imagesgallery.py]: Done

File: imagesgallery/imagesgallery.py [hunk 3]
```diff
@@ -74,22 +96,56 @@ def student_view(self, context=None):
         if context:
             pass  # TO-DO: do something based on the context.
 
-        # Remove the bundle default generated by webpack
-        script_to_remove = r'<script defer="defer" src="bundle.js"></script>'
+        xblock_id = self.block_id
+        # XBlock IDs typically follow this format: block-v1:{course_name}+type@{xblockname}+block@{id_xblock}
+        # This code extracts the last part of the string, which is {id_xblock}
+        xblock_id_splitted = self.block_id_parsed
```
Comment (BryanttV) [inline imagesgallery/imagesgallery.py]: Can you delete this variable and directly access the attribute where you need it?
Same for the `xblock_id` variable above

Comment (johnvente) [reply imagesgallery/imagesgallery.py]: Yep, done

**Comment by BryanttV** [comment]
LGTM. Thanks!

**Review by BryanttV — APPROVED** [review]

**Pull request merged**